src/components/layout/CyVerseAppBar.js

```diff
@@ -642,104 +613,81 @@ function CyverseAppBar(props) {
-                        <>
-                            <Divider />
-                            {adminDrawerItems}
-                        </>
-                    )}
-                </Drawer>
-            </Hidden>
-            <Hidden only={["sm", "md", "lg", "xl"]}>
-                <Drawer
-                    className={clsx(classes.drawer, {
-                        [classes.drawerOpen]: open,
-                        [classes.drawerClose]: !open,
-                    })}
-                    classes={{
-                        paper: clsx({
-                            [classes.drawerOpen]: open,
-                            [classes.drawerClose]: !open,
-                        }),
-                    }}
-                    open={open}
-                    onClose={toggleDrawer(false)}
-                >
+                </Hidden>
+                <Hidden smUp>
                     <div
                         id={build(ids.DRAWER_MENU, ids.ACCOUNT_MI)}
                         style={{ margin: 8 }}
                     >
-                        <UserMenu
-                            baseId={build(ids.DRAWER_MENU, ids.ACCOUNT_MI)}
-                            profile={userProfile}
-                            onLogoutClick={onLogoutClick}
-                            onManageAccountClick={() =>
-                                window.open(
-                                    "https://user.cyverse.org",
-                                    "_blank"
-                                )
-                            }
-                        />
+                        {userProfile ? (
+                            <UserMenu
+                                baseId={build(ids.DRAWER_MENU, ids.ACCOUNT_MI)}
+                                profile={userProfile}
+                                onLogoutClick={onLogoutClick}
+                                onManageAccountClick={() =>
+                                    window.open(
+                                        constants.CYVERSE_USER_PORTAL,
+                                        "_blank"
+                                    )
+                                }
+                            />
+                        ) : (
+                            accountAvatar
+                        )}
                     </div>
-                    <Divider />
-                    {drawerItems}
-                    {adminUser && (
-                        <>
-                            <Divider />
-                            {adminDrawerItems}
-                        </>
-                    )}
-                </Drawer>
-            </Hidden>
+                </Hidden>
+                <Divider />
+                {drawerItems}
+                {(!isXsDown || open) && adminUser && (
+                    <>
+                        <Divider />
+                        {adminDrawerItems}
+                    </>
+                )}
+            </Drawer>
```

Comment:
This whole thing is basically reorganizing: in both setups, we have a drawer with various props, some conditional stuff at the top depending on the screen size, then the main drawer items, then the admin drawer (with slightly different conditions due to the different drawer types). This helps the code size some since we're sharing more, and I think it also makes the shared vs. differing code more obvious -- and if I understand React properly, this should mean less DOM manipulation too. As an additional bonus, having more things outside of `Hidden` components means that more can be server-side rendered.